framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -400,6 +404,11 @@ protected void addCheckedCodeDefaults(QualifierDefaults defs) {
                                         + "@DefaultQualifierInHierarchy or @DefaultFor(DefaultLocation.OTHERWISE) not found. "
                                         + "Every checker must specify a default qualifier.");
         }
+        Set<? extends AnnotationMirror> tops = this.qualHierarchy.getTopAnnotations();
```

Comment:
As tops and bottoms is only used in the condition, I would move their initialization there and limit their scope.